maven-plugin-plugin/src/main/java/org/apache/maven/plugin/plugin/DescriptorGeneratorMojo.java

```diff
@@ -360,8 +346,8 @@ public void generate() throws MojoExecutionException {
             request.setEncoding(encoding);
             request.setSkipErrorNoDescriptorsFound(skipErrorNoDescriptorsFound);
             request.setDependencies(filterMojoDependencies());
-            request.setLocal(this.local);
-            request.setRemoteRepos(this.remoteRepos);
+            request.setRemoteRepos(project.getRemoteProjectRepositories());
+            request.setRepoSession(repoSession);
```

Comment:
Stupid question: If we already pass the project in the ctor why then pass `project.getRemoteProjectRepositories()` which can be retrieved internally?